codalab/lib/download_manager.py

```diff
@@ -52,25 +61,24 @@ def get_target_info(self, uuid, path, depth):
         worker.download_util.get_target_info.
         """
         try:
-            bundle_state = self._bundle_model.get_bundle_state(uuid)
+            bundle_state = self.get_bundle_state(uuid)
         except NotFoundError:
             bundle_state = None
 
         # Return None if invalid bundle reference
         if bundle_state is None:
             return None
-        elif bundle_state != State.RUNNING:
-            bundle_path = self._bundle_store.get_bundle_location(uuid)
+
+        if self._is_available_locally(uuid):
+            bundle_path = self.get_bundle_location(uuid)
             try:
-                return download_util.get_target_info(bundle_path, uuid, path, depth)
+                key = (bundle_path, uuid, path, depth)
+                return self._cached_if_stable(uuid, 'target_info', key, lambda: download_util.get_target_info(*key))
             except download_util.PathException as e:
                 raise UsageError(e.message)
         else:
-            # get_target_info calls are sent to the worker even on a shared file
-            # system since 1) due to NFS caching the worker has more up to date
-            # information on directory contents, and 2) the logic of hiding
-            # the dependency paths doesn't need to be re-implemented here.
-            worker = self._worker_model.get_bundle_worker(uuid)
+            # TODO: validate that this is a reasonable approach
```

Comment:
We don't need to change this branch since all our bundles should be available locally, right?